Modules/_ctypes/stgdict.c

```diff
@@ -302,7 +302,15 @@ MakeAnonFields(PyObject *type)
             Py_DECREF(anon_names);
             return -1;
         }
-        assert(Py_TYPE(descr) == &PyCField_Type);
+        if (Py_TYPE(descr) != &PyCField_Type) {
+            PyErr_Format(PyExc_AttributeError,
+                         "'%U' is specified in _anonymous_ but not in "
```

Comment:
What if `fname` is not a string?

Reply:
IIUC, `PyObject_GetAttr()` would return NULL